app/indexing/index_documents.py

```diff
@@ -68,7 +74,10 @@ def index_documents(documents: List[BasicDocument]):
             db_documents = []
             for document in documents:
                 # Split the content into paragraphs that fit inside the database
-                paragraphs = Indexer._split_into_paragraphs(document.content)
+                if document.file_type != FileType.PDF:
+                    paragraphs = Indexer._split_into_paragraphs(document.content)
+                else:
+                    paragraphs = split_PDF_into_paragraphs(document.content)
```

Comment:
now when content is always simple str, is this needed?

Reply:
@Roey7 yes because the way i split documents is different than the other one , `split_PDF_into_paragraphs` is different than this ` Indexer._split_into_paragraphs` so i have to check the file type

Reply:
@Roey7 with your above suggestion to move the `split_PDF_into_paragraphs ` function to the pdf parser module it is important to do this check

Reply:
@Roey7 reverted back after discussions